### ISSUE TITLE
UnleashClientFactory and custom strategies

### ISSUE THREAD

**Issue by nickcox** [Unleash/unleash-client-dotnet#61]
Not sure if this is a feature request or just something I've missed in the API but I'm struggling to register a custom strategy when creating an instance of the client via `UnleashClientFactory`. Both of the `CreateClient...` methods seem to be hard-coded with `Array.Empty<IStrategy>()`.

My use case is that I would like to create the client synchronously with a custom strategy. Any suggestions on the best way to achieve this?

**Comment by ivarconr** [comment]
The standard way of initializing Unleash synchronously would simply be:

```c#
var settings = new UnleashSettings()
{
  AppName = "dot-net-client",
  Environment = "local",
  UnleashApi = new Uri("https://app.unleash-hosted.com/demo/api/"),
  CustomHttpHeaders = new Dictionary<string, string>()
  {
    {"Authorization","56907a2fa53c1d16101d509a10b78e36190b0f918d9f122d" }
  }
};

var unleash = new DefaultUnleash(settings, false, <custom strategies>);
```

**Comment by nickcox** [comment]
Oh, right. I was under the impression that would schedule a feature fetch to run asynchronously. If I attempt to use the client  immediately after...

```c#
var unleash = new DefaultUnleash(settings, false, <custom strategies>);
```

... it doesn't seem to be aware of any features for a second or so. Whereas, if I use [`UnleashClientFactory.CreateClient(settings, true)`](https://github.com/Unleash/unleash-client-dotnet/blob/f2f6ce9046614647d06a44191146568b3aeaa63d/src/Unleash/ClientFactory/UnleashClientFactory.cs#L19), it blocks until the features have been fetched but I can't seem to specify custom strategies.

**Comment by ivarconr** [comment]
> ... would schedule a feature fetch to run asynchronously

That's correct (I misunderstood your question), by default the client SDK will perform the fetching in the background, but not do any initial blocking for the first request. 

But yes, you are correct the factory creates does not allow you to inject custom strategies. You should be able to write your own custom factory, but we will also add this to the factory methods, to facilitate it better.

**Comment by nickcox** [comment]
Thanks, Ivar. I did have a crack at writing my own factory but it seemed that everything I would need to make it happen is `internal`. Any clues much appreciated. 😄

**Comment by ivarconr** [comment]
fixed in 1.6.0
